resources/tentacle.rb

```diff
@@ -19,28 +19,211 @@
 # limitations under the License.
 #
 
-actions :install, :configure, :register, :remove, :uninstall
+include OctopusDeploy::Tentacle
+
+resource_name 'octopus_deploy_tentacle'
+
+property :instance, kind_of: String, default: 'Tentacle'
+property :version, kind_of: String
+property :checksum, kind_of: String
+property :home_path, kind_of: String, default: 'C:\Octopus'
+property :config_path, kind_of: String, default: 'C:\Octopus\Tentacle.config'
+property :app_path, kind_of: String, default: 'C:\Octopus\Applications'
+property :trusted_cert, kind_of: String
+property :polling, kind_of: [TrueClass, FalseClass], default: false
+property :configure_firewall, kind_of: [TrueClass, FalseClass], default: false
+property :port, kind_of: [Integer, NilClass], default: nil
+property :cert_file, kind_of: String, default: 'C:\Octopus\tentacle_cert.txt'
+property :upgrades_enabled, kind_of: [TrueClass, FalseClass], default: true
+property :server, kind_of: String
+property :api_key, kind_of: String
+property :roles, kind_of: Array
+property :environment, kind_of: [String, Array], default: node.chef_environment
+property :tenants, kind_of: Array, default: nil
+property :tenant_tags, kind_of: Array, default: nil
+property :tentacle_name, kind_of: String, default: node.name
+property :service_user, kind_of: String, default: nil
+property :service_password, kind_of: String, default: nil
+property :public_dns, kind_of: String, default: node['fqdn']
```

Comment:
`node.x` vs `node['x']` is there a standard way of choosing one over the other, or a reason for not being consistent? I think both works